lib/collection/src/grouping/aggregator.rs

```diff
@@ -95,20 +94,24 @@ impl GroupsAggregator {
             }
 
             // Insert score if better than the group best score
-            self.group_best_scores
+            self.groups_best_point
                 .entry(group_key.clone())
-                .and_modify(|e| {
-                    *e = match self.order {
-                        Order::LargeBetter => point.score.max(*e),
-                        Order::SmallBetter => point.score.min(*e),
+                .and_modify(|p_id| {
+                    let other_score: &ScoredPoint = self.groups[&group_key].get(p_id).unwrap();
```

Comment:
Having to dereference the point id from the groups and then unwrapping makes the logic to be more complicated than it should, IMO. 

If we instead store the actual `ScoredPoint` as a value in the `groups_best_point` hashmap, we won't need any dereferencing, and it wouldn't hurt memory too much, since it is only one per group.